File: LeetCode/Study Plan/Practice Problems/Easy/Contains Duplicate/Java/containsDuplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashMap<Integer, Integer> count = new HashMap<>();
        for (int val : nums) {
            count.put(val, count.getOrDefault(val, 0)+1);
        }
        return count.size() != nums.length;
    }
}

File: LeetCode/neetcode 150/arrays and hashing/contains_duplicate.cc
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_map<int, int> hash_map;
        for (int num : nums) {
            hash_map[num]++;
            if (hash_map[num] > 1) return true;
        }
        return false;
    }
};
